### DISCUSSION TITLE
aws_codepipeline_actions: not able to access outputs/variable of CloudFormationCreateUpdateStackAction

### DISCUSSION THREAD

**Discussion by entest-hai** [aws/aws-cdk#23166] [Q&A]
### Describe the bug

This is application stack 
```tsx
... 
 this.url = new CfnOutput(this, "Url", {
      description: "api url",
      exportName: "Url",
      value: api.url,
    });
```

This is a deploy action 
```tsx
 const deployPreProd =
      new aws_codepipeline_actions.CloudFormationCreateUpdateStackAction({
        actionName: "DeployPreProdApplication",
        templatePath: cdkBuildOutput.atPath("ApplicationStack.template.json"),
        stackName: "PreProductApplicationStack",
        adminPermissions: true,
        variablesNamespace: "PreProdVariables",
        outputFileName: "PreProdOutputs",
        output: preProdOutput,
      });
```
Then I want to access a output (API URL) from the deployed stack and pass to a codebuild action 
```tsx
 const integtestBuildAction = new aws_codepipeline_actions.CodeBuildAction({
      environmentVariables: {
        SERVICE_URL: {
          value: "HOW TO ACCESS THE API URL HERE? FROM THE DEPLOYED STACK ABOVE",
        },
      },
      actionName: "IntegTest",
      project: integtestCodeBuild,
      input: sourceOutput,
    });
```

[The CDK document](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_codepipeline_actions-readme.html) does not describe how CloudFormationCreateUpdateStackAction emit outputs/variable? 

### Expected Behavior

CloudFormationCreateUpdateStackAction should emit outputs/variables which are accessible to subsequent actions 

### Current Behavior

Not able to access outputs/variable from CloudFormationCreateUpdateStackAction. The only way to work around is to using boto3, read the deployed stack (describe) and looking for the outputs given exported names. 

### Reproduction Steps

Just reuse the code above 

### CDK CLI Version

2.27.0

### Node.js Version

16.15.1

### OS

Amazon Linux 2

### Language

Typescript

**Comment by indrora** [comment]
if I understand your problem correctly, you are

* Deploying an HTTP API
* Deploying a build + test using CodePipeline

You'll need to use Parameter Store to glue the two together: When the CodePipeline resource is generated, the HTTP API has yet to be fully deployed, and it won't know it until the whole CloudFormation stack has been reified.

**Comment by peterwoodworth** [comment]
> The only way to work around is to using boto3, read the deployed stack (describe) and looking for the outputs given exported names.

When you need to make an SDK call to get some information, the best thing to do is to use a custom resource. Specifically, our helper construct [AwsCustomResource](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.custom_resources.AwsCustomResource.html) makes it easy to make any arbitrary SDK call.

The suggestion above with parameter store is also a solution. But I'm also wondering why the output artifact from this action cannot be used as an input for a later step?

I think this topic will get better visibility as a discussion, converting now.
